cookbook/security/entity_provider.rst

```diff
@@ -202,72 +164,33 @@ interface forces the class to implement the five following methods:
-
-.. sidebar:: What is the importance of serialize and unserialize?
-
-    The :phpclass:`Serializable` interface and its ``serialize`` and ``unserialize``
-    methods have been added to allow the ``User`` class to be serialized
-    to the session. This may or may not be needed depending on your setup,
-    but it's probably a good idea. The ``id`` is the most important value
-    that needs to be serialized because the
-    :method:`Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider::refreshUser`
-    method reloads the user on each request by using the ``id``. In practice,
-    this means that the User object is reloaded from the database on each
-    request using the ``id`` from the serialized object. This makes sure
-    all of the User's data is fresh.
-
-    Symfony also uses the ``username``, ``salt``, and ``password`` to verify
-    that the User has not changed between requests. Failing to serialize
-    these may cause you to be logged out on each request. If your User implements
-    :class:`Symfony\\Component\\Security\\Core\\User\\EquatableInterface`,
-    then instead of these properties being checked, your ``isEqualTo`` method
-    is simply called, and you can check whatever properties you want. Unless
-    you understand this, you probably *won't* need to implement this interface
-    or worry about it.
-
-Below is an export of the ``User`` table from MySQL with user ``admin`` and
-password ``admin`` (which has been encoded). For details on how to create
-user records and encode their password, see :ref:`book-security-encoding-user-password`.
+To learn more about each of these, see :class:`Symfony\\Component\\Security\\Core\\User\\UserInterface`.
 
-.. code-block:: bash
+What do the serialize and unserialize Methods do?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    $ mysql> SELECT * FROM acme_users;
-    +----+----------+------------------------------------------+--------------------+-----------+
-    | id | username | password                                 | email              | is_active |
-    +----+----------+------------------------------------------+--------------------+-----------+
-    |  1 | admin    | d033e22ae348aeb5660fc2140aec35850c4da997 | admin@example.com  |         1 |
-    +----+----------+------------------------------------------+--------------------+-----------+
-
-The next part will focus on how to authenticate one of these users
-thanks to the Doctrine entity user provider and a couple of lines of
-configuration.
+At the end of each request, the User object is serialized to the session.
+On the next request, it's unserialized. To help PHP do this correctly, you
+need to implement ``Serializable``. But you don't need to serialize everything:
+you only need a few fields (the ones shown above plus a few extra if you
+decide to implement :ref:`AdvancedUserInterface <security-advanced-user-interface>`).
+On each request, the ``username`` is used to query for a fresh ``User`` object
```

Comment:
not actually true. The Entity user provider actually uses the primary key for that to avoid a security issue in case the username is mutable (see the security announcement for Symfony 2.0.6)